pkg/ipamplugin/ipam_plugin.go

```diff
@@ -207,12 +207,16 @@ func cmdAdd(args *skel.CmdArgs) error {
 
 		logger.Infof("Calico CNI IPAM request count IPv4=%d IPv6=%d", num4, num6)
 
-		v4pools, err := utils.ResolvePools(ctx, calicoClient, conf.IPAM.IPv4Pools, true)
+		ctx2, cancel := context.WithTimeout(ctx, 10*time.Second)
```

Comment:
I think we should just make ctx be a `context.WithTimeout`, I think this whole cmdAdd function could use that one context and should all be done in 10 seconds.
I think it would be reasonable to make the same changes in the cmdDel.

Reply:
Fixed, please take a look.